### ISSUE TITLE
thread 'rustc' panics

### ISSUE THREAD

**Issue by Milo123459** [rust-lang/rust-clippy#7735]
I tried this code:

```rust
<code>
```

### Meta

https://github.com/serenity-rs/serenity/runs/3735636478?check_suite_focus=true

Here is a CI run, i sadly can't get that information.

**Comment by Milo123459** [comment]
```rs
src/tools/clippy/clippy_lints/src/implicit_hasher.rs:175:34
```
That's where it fails apparently.

**Comment by Milo123459** [comment]
Started getting this locally:
```rs
warning: this `impl` can be derived
  --> src\cache\settings.rs:22:1
   |
22 | / impl Default for Settings {
23 | |     fn default() -> Self {
24 | |         Settings {
25 | |             max_messages: usize::default(),
26 | |         }
27 | |     }
28 | | }
   | |_^
   |
   = note: `#[warn(clippy::derivable_impls)]` on by default
   = help: try annotating `cache::settings::Settings` with `#[derive(Default)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls

thread 'rustc' panicked at 'failed to create span for type parameters', src\tools\clippy\clippy_lints\src\implicit_hasher.rs:175:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.1.57 (98c86195 2021-09-27)

query stack during panic:
#0 [analysis] running analysis passes on this crate
end of query stack
warning: `serenity` (lib) generated 1 warning
error: could not compile `serenity`; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
thread 'rustc' panicked at 'failed to create span for type parameters', src\tools\clippy\clippy_lints\src\implicit_hasher.rs:175:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.1.57 (98c86195 2021-09-27)

query stack during panic:
#0 [analysis] running analysis passes on this crate
end of query stack
warning: `serenity` (lib test) generated 1 warning (1 duplicate)
error: build failed
```
`rustc -vV`
```
rustc 1.57.0-nightly (98c861950 2021-09-27)
binary: rustc
commit-hash: 98c8619502093f34ca82f8f26ccf32e753924440
commit-date: 2021-09-27
host: x86_64-pc-windows-msvc
release: 1.57.0-nightly
LLVM version: 13.0.0
```

**Comment by giraffate** [comment]
Thanks for the report!

I'm closing this because this is duplicate of https://github.com/rust-lang/rust-clippy/issues/7712.

**Comment by Milo123459** [comment]
Ah I see, sorry
